### PR TITLE
DAOS-6726 array: fixing memleak in FI testing

### PR DESCRIPTION
**Pull request by mchaarawi**
Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/array/dc_array.c [hunk 1]
```diff
@@ -1770,7 +1773,7 @@ dc_array_io(daos_handle_t array_oh, daos_handle_t th,
 	 */
 	if (op_type == DAOS_OPC_ARRAY_READ && array->byte_array) {
 		if (head == NULL) {
-			tse_task_complete(stask, rc);
+			tse_task_complete(stask, rc = 0);
```
Comment (ashleypittman) [inline src/client/array/dc_array.c:1776]: This line looks a little odd, what exactly is it trying to do?  Mind you rc isn't used later in this function so perhaps just pass 0 to the function?
```suggestion
			rc = 0;
			tse_task_complete(stask, rc);
```

**Review by ashleypittman — COMMENTED** [review]
100 chars per line does work a lot better.

**Review by ashleypittman — APPROVED** [review]

File: src/client/array/dc_array.c [hunk 2]
```diff
@@ -871,9 +873,8 @@ dc_array_close(tse_task_t *task)
 	/** Create task to close object */
 	rc = daos_task_create(DAOS_OPC_OBJ_CLOSE, tse_task2sched(task),
 			      0, NULL, &close_task);
-	if (rc != 0) {
-		D_ERROR("Failed to create object_close task "DF_RC"\n",
-			DP_RC(rc));
+	if (rc) {
```
Comment (dmiter) [inline src/client/array/dc_array.c:876]: Same style for all instancies.
```suggestion
	if (rc != 0) {
```

**Review by dmiter — APPROVED** [review]
Thanks for this fix.

File: src/client/array/dc_array.c (hunk 2 above)
Comment (mchaarawi) [reply src/client/array/dc_array.c:876]: unfortunately we mix those a lot everywhere. we can create a ticket to address this separately.

**Pull request merged**